index/scorch/persister.go

```diff
@@ -553,10 +553,11 @@ func (s *Scorch) persistSnapshotDirect(snapshot *IndexSnapshot) (err error) {
 			}
 		}()
 		for segmentID, path := range newSegmentPaths {
-			newSegments[segmentID], err = zap.Open(path)
```

Comment:
I guess the `defer` loop above already ignores `nil` so this probably is fine as-is.